examples/user_guide/10-Transforming_Elements.ipynb

```diff
@@ -298,6 +298,105 @@
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%output backend=\"matplotlib\"\n",
+    "\n",
+    "from scipy import ndimage\n",
+    "\n",
+    "class image_filter(hv.Operation):\n",
+    "    \n",
+    "    sigma = param.Number(default=5)\n",
+    "    \n",
+    "    type_ = param.String(default=\"low-pass\")\n",
+    "\n",
+    "    def _process(self, element, key=None):\n",
+    "        # we don't need the x and y coordinates in this example, \n",
+    "        # but we could access them like this\n",
+    "        x_coords = element.dimension_values(0, expanded=False)\n",
+    "        y_coords = element.dimension_values(1, expanded=False)\n",
+    "        \n",
+    "        # setting flat=False will preserve the matrix shape\n",
+    "        data = element.dimension_values(2, flat=False)[::-1]\n",
+    "        \n",
+    "        if self.p.type_ == \"high-pass\":\n",
+    "            new_data = data - ndimage.gaussian_filter(data, self.p.sigma)\n",
+    "        else:\n",
+    "            new_data = ndimage.gaussian_filter(data, self.p.sigma)\n",
+    "            \n",
+    "        # make an exact copy of all setting and axes etc., just with different data:\n",
+    "        element = element.clone(data=new_data)\n",
+    "        element = element.relabel(element.label + \" ({} filtered)\".format(self.p.type_))\n",
+    "        return element\n",
+    "\n",
+    "stairs_map = hv.HoloMap({sigma: image_filter(stairs_image, sigma=sigma)\n",
+    "                         for sigma in range(0, 12, 1)}, kdims=\"sigma\")\n",
+    "stairs_map"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Just as in the previous example, it is quite straight-forward to build a HoloMap containing results for different parameter values. Inside the ``_process()`` method, the given parameters can be accessed as ``self.p.<parameter-name>`` (note that ``self.<parameter_name>`` always contains the default value!). Since we did not specify the ``_type`` parameter, it defaulted to \"low-pass\".\n",
+    "\n",
+    "There are some peculiarities when applying operations to two-dimensional elements:\n",
+    "\n",
+    "- Understanding the ``dimension_values()`` method: In principle, one could use ``element.data`` to access the element's data, however, using ``dimension_values()`` means one does not have to worry about different data formats for different elements. The first parameter specifies the dimension to be returned. Values 0 and 1 refer to the horizontal and vertical axis, respectively. For  ``Image`` and ``Raster`` elements, 2 refers to the magnitude, while for ``RGB``, 2, 3 and 4 refer to the red, green and blue channels. Setting ``expanded=False`` yields only the axis, while the default setting ``expanded=True`` returns a value for every pixel. Specifying ``flat=False`` means that the data's matrix shape will be preserved, which is what we need for this kind of filter.\n",
```

Comment:
Some things that should maybe be clarified:

1. While it is true that the data format depends on the element type, all elements also support a variety of different formats, which are defined by the ``datatype``. In the background a number of Interface classes allow elements to access and manipulate the data in different formats, be that numpy arrays, dask arrays, xarrays or whatever else. We need to add a user guide about interfaces eventually so maybe just say something like: "In principle, one could use ``element.data`` to access the element's data, however, since HoloViews can wrap a wide range of data formats ``dimension_values()`` provides an API that lets you access the data without having to worry about the type of the data.

2. While it is true that ``dimension_values`` allows using integers to reference which values you want, it is primarily about the dimensions. On a 2D Element 0 and 1 correspond to the two key dimensions, while subsequent values refer to the value dimensions, basically it's defined as ``element.dimensions()[index]``. So maybe say something like:

On a 2D element like an ``Image`` or ``Raster`` the first two dimensions reference the key dimensions, so passing an index of 0 or 1 will return the x- and y-axis values respectively. Any subsequent dimensions will be value dimensions, e.g. on an Image index value 2 will refer to the intensity values and on an ``RGB`` index values 2, 3, and 4 will return the  Red, Green and Blue intensities instead.